src/connectors/kessel/impl.js

```diff
@@ -0,0 +1,205 @@
+        try {
+            // Create the check request using the new API structure
+            const checkRequest = {
+                subject: {
+                    resource: {
+                        reporter: {
+                            type: "rbac"
+                        },
+                        resourceId: `${this.kesselConfig.principalDomain}/${userId}`,
+                        resourceType: "principal"
+                    }
+                },
+                object: {
+                    reporter: {
+                        type: "rbac"
+                    },
+                    resourceId: workspaceId,
+                    resourceType: "workspace"
+                },
+                relation: relation
+            };
+
+            // Use async/await with the new client
+            const response = await this.kesselClient.check(checkRequest);
+            return response.allowed || false;
+        } catch (error) {
+            log.warn({ error, userId, workspaceId, relation }, 'Kessel check call failed');
+            throw error;
+        }
+    }
+
+    extractWorkspaceId(identity) {
+        // Extract workspace/organization ID from identity
+        return identity.identity?.account_number ||
+               identity.identity?.org_id ||
+               identity.account_number ||
+               identity.org_id ||
+               'default';
+    }
+
+    getIdentityFromHeaders() {
+        const headers = this.getForwardedHeaders();
+
+        // Try to extract identity from x-rh-identity header
+        if (headers['x-rh-identity']) {
+            try {
+                const identityHeader = Buffer.from(headers['x-rh-identity'], 'base64').toString();
+                return JSON.parse(identityHeader);
+            } catch (error) {
+                log.warn({ error }, 'Failed to parse x-rh-identity header');
+            }
+        }
+
+        return null;
+    }
+
+    // Compatibility method to check specific permission
+    async hasPermission(resource, action, subject_org_id) {
+        if (!this.kesselConfig.enabled || !this.initialized || !this.kesselClient) {
+            return false;
+        }
+
+        const startTime = Date.now();
+        try {
+            // Convert traditional RBAC permission to workspace permission
+            const workspacePermission = this.convertRbacToWorkspacePermission(resource, action);
+            if (!workspacePermission) {
+                log.warn({ resource, action }, 'Unknown permission mapping');
+                return false;
+            }
+
+            // Get workspace ID from subject_org_id or use default
+            const workspace = await this.getDefaultWorkspaceIdForSubject(subject_org_id);
+            const workspaceId = workspace?.id || null;
+
+            // Check the specific permission using async/await
+            const allowed = await this.checkSinglePermission(subject_org_id, workspaceId, workspacePermission);
```

Comment:
Almost missed this.

We actually need the `user_id` - not the org_id here- You can extract it from the identity like so:

```
identity = the_json_x_rh_identity_header['identity'];
if (identity['type'] === 'ServiceAccount') {
  user_id = identity['service_account']['user_id']
} else if (identity['type'] === 'User') {
  user_id = identity['user']['user_id']
} else {
  throw new Error('unknown identity type:' + identity['type']);
}

org_id = identity['org_id']
```

Reply:
I had used `org_id` here because it looked like the `fetchWorkspace` and `fetchDefaultWorkspace` functions from the SDK were using `orgId`.

- https://github.com/project-kessel/kessel-sdk-node/blob/main/src/kessel/rbac/v2.ts#L61

Reply:
I have made those changes. Let me know if that resolves this issue.